python/lsst/obs/base/_fitsRawFormatterBase.py

```diff
@@ -416,3 +376,21 @@ def observationInfo(self):
             self._observationInfo = ObservationInfo(self.metadata, translator_class=self.translatorClass,
                                                     filename=path)
         return self._observationInfo
+
+    def attachComponentsFromMetadata(self, exposure):
+        """Attach all `lsst.afw.image.Exposure` components derived from
+        metadata (including the stripped metadata itself).
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to attach components to (modified in place).  Must already
+            have a detector attached.
+        """
+        info = exposure.getInfo()
+        info.setFilterLabel(self.makeFilterLabel())
+        info.setVisitInfo(self.makeVisitInfo())
+        info.setWcs(self.makeWcs(info.getVisitInfo(), info.getDetector()))
+        # We don't need to call stripMetadata() here because it has already
+        # been stripped during creation of the ObservationInfo, WCS, etc.
```

Comment:
ObservationInfo shouldn't be stripping anything. WCS is the only thing that does and I think only if we call the read-from-header variant and not the read-from-visitinfo variant. Strip metadata wastes its time and calls makeVisitInfo but does strip the header WCS -- maybe I'm getting confused but can you check that this header has been stripped?

Reply:
I'll drop `ObservationInfo` from the comment as per the other thread.

The default `makeWcs` implementation definitely does strip, because it starts by making the WCS from metadata first - both as a fallback in case there's no `VisitInfo` for the boresight+cameraGeom one, and to ensure that the stripping happens.  But I should at least go add a note to the documentation that this is an expected side effect of any reimplementation of `makeWcs`; it's unfortunate that we have to rely on side effects at all, but given that the real work is happening in an afw routine via side effects, I don't think we have much choice.